nomad/leader_test.go

```diff
@@ -1234,21 +1244,34 @@ func TestLeader_RollRaftServer(t *testing.T) {
 		c.RaftConfig.ProtocolVersion = 3
 	})
 	defer cleanupS4()
-	TestJoin(t, s4, s2)
+	TestJoin(t, s2, s3, s4)
 	servers[0] = s4
 
 	// Kill the second v2 server
 	s2.Shutdown()
 
 	for _, s := range []*Server{s3, s4} {
-		retry.Run(t, func(r *retry.R) {
+		s.RemoveFailedNode(s2.config.NodeID)
+
+		retry.RunWith(&retry.Counter{
+			Count: int(10 * testutil.TestMultiplier()),
+			Wait:  time.Duration(testutil.TestMultiplier()) * time.Second,
+		}, t, func(r *retry.R) {
```

Comment:
I made the retry fairly large because it can take a while for the peer to be removed. Hopefully this will prevent this test from being flaky.